### PR TITLE
Week3assesment

### PR DESCRIPTION
**Pull request by Wagnergon**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ✅ ] Coding Challenges
- [ ✅ ] Interview Questions
- [ ✅ ] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by Wagnergon** [comment]
I messed up pushing everything up, I may have pushed everything up to the main branch on accident.

File: code-challenges.test.js [hunk 1]
```diff
@@ -22,11 +40,47 @@ const fibLength2 = 10
...
+}
+
+// the test did not pass 
+// // Test Suites: 1 failed, 1 total
+// Tests:       0 total
+// Snapshots:   0 total
+// Time:        1.103 s
+// Ran all test suites.
+
+// im not sure why the test isnt passing everything seems right to me.
```
Comment (DFWilliams) [inline code-challenges.test.js:68]: Your code looks great! I would encourage you to look at your test, theres a couple of different function names.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,6 +14,24 @@
...
+// Psuedocode
+// Write out a describe statemtn that contains the function fibanacci
+// create an it statement describing that itll take in the variable and put out a sequence.
+// create two expect statements using .toEqual to tell the test what the output should equal 
+
+describe ("fibanacci", () => {
+  it ("should return the expected fibanacci seuquence", () => {
+    const fibLength1 = 6
+    const fibLength2 = 10
+      expect (fib(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
```
Comment (DFWilliams) [inline code-challenges.test.js:26]: You have a couple of different function names in your test. We want them to be the same so we don't run into incorrect failures

File: code-challenges.test.js (hunk 1 above)
Comment (DFWilliams) [inline code-challenges.test.js:68]: Your code looks great! Take a look at the note I gave in your test.

File: code-challenges.test.js [hunk 3]
```diff
@@ -65,3 +184,28 @@ const accountTransactions3 = []
...
+const sumArrays = (array) => {
+  let sum = 0
+  return array.map((value) => {
+    sum += value
+    return sum
+  })
+}
+
+
+// i re ran my test and the first two passed but my last one on the sum of the arrays did not
```
Comment (DFWilliams) [inline code-challenges.test.js:206]: Great approach! Take a look at your test.

File: interview-questions.md [hunk 4]
```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Your answer:
+Your answer: Yarn are the files you use when you want to test youre javascript code.
 
-Researched answer:
+Researched answer: Yarn is a package manager developed by facebook. It modifies the files where the project is that you are working on. It adds node-modules in your directory.
 
 3. What is a React component?
 
-Your answer:
+Your answer: It is the reusable logic in youre react app. For example when we did the colorbox app we made a component that was the logic for the colorbox changing colors.
```
Comment (DFWilliams) [inline interview-questions.md:23]: I like that you added an example to drive home your point!

File: react-debugging.html [hunk 5]
```diff
@@ -63,13 +63,16 @@
...
+// its missing a div
         return (
+          <div>
           <h1>Draw a Card</h1>
-          <button onClick={}>
+          <button onClick={drawCard}> 
             Click to Draw a Card
           </button>
-          <button onclick={shuffle}>
+          <button onClick={shuffle}>
```
Comment (DFWilliams) [inline react-debugging.html:75]: Nailed It! Good Job

**Review by DFWilliams — APPROVED** [review]
Wonderful work this week Wags! I can see improvement. This displays a Novice comprehension level, but you are on the cusp of Adv. Beginner. Some things I think would help you to get to that next level is 1. In your interview questions, I encourage you to use examples any chance you get. It helps solidify your understanding of the concepts. 2. Your code is looking great, but when writing your test be aware of how you name your functions in your describe statement and in your expect statements. Great job!

**Pull request merged**